File: solutions/alfi.compitest/source_gen/alfi/compitest/Cars/Vehicle.cs
/// <filename>
///     Vehicle.cs
/// </filename>
using Alf.Library;

namespace Cars
{
    public abstract class Vehicle
    {
        public virtual void drive()
        {
            this.distance = this.distance + 1;
        }

        protected virtual int getSpeed()
        {
            return 5;
        }

        protected virtual void defaultStartEngine()
        {
            BasicInputOutput.WriteLine("Vehicle engine started");
        }

        public abstract void startEngine();

        public int distance = 0;
    }
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/Cars/Vehicle.cpp
#include "Vehicle.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>

namespace Cars
{
  void Vehicle::drive()
  {
    this->distance = this->distance + 1;
  }
  int Vehicle::getSpeed()
  {
    return 5;
  }
  void Vehicle::defaultStartEngine()
  {
    alf::library::BasicInputOutput::WriteLine(std::string("Vehicle engine started"));
  }
}
